### PR TITLE
FISH-6430: skipping BdId TransactionalExtension type to not generate error logs

### PR DESCRIPTION
**Pull request by breakponchito**
## Description

This is a fix to prevent error logs related to inject TransactionalScopedCDIEventHelperImpl for TransactionScoped CDI Beans
### Testing Performed

local testing using the attached reproducer (adjusted to be compatible with most recent version of jakarta) from the ticket: [FISH-6430](https://payara.atlassian.net/browse/FISH-6430)

logs looks more cleaned:
`[#|2022-09-27T12:01:10.483-0500|INFO|Payara 6.2022.1.Alpha5-SNAPSHOT|javax.enterprise.system.core|_ThreadID=145;_ThreadName=admin-thread-pool::admin-listener(1);_TimeMillis=1664298070483;_LevelValue=800;|
  simple-ear was successfully deployed in 3,935 milliseconds.|#]

[#|2022-09-27T12:02:04.044-0500|INFO|Payara 6.2022.1.Alpha5-SNAPSHOT||_ThreadID=131;_ThreadName=http-thread-pool::http-listener-1(3);_TimeMillis=1664298124044;_LevelValue=800;|
  ServiceBean::doWork() called|#]

[#|2022-09-27T12:02:10.350-0500|INFO|Payara 6.2022.1.Alpha5-SNAPSHOT||_ThreadID=131;_ThreadName=http-thread-pool::http-listener-1(3);_TimeMillis=1664298130350;_LevelValue=800;|
  **TestTransactionalScopeBean initialized**. Value is 1664298130350|#]

[#|2022-09-27T12:02:59.559-0500|INFO|Payara 6.2022.1.Alpha5-SNAPSHOT||_ThreadID=131;_ThreadName=http-thread-pool::http-listener-1(3);_TimeMillis=1664298179559;_LevelValue=800;|
  ServiceBean:: testTransactionalScope value is 1664298130350|#]

[#|2022-09-27T12:03:17.772-0500|INFO|Payara 6.2022.1.Alpha5-SNAPSHOT||_ThreadID=131;_ThreadName=http-thread-pool::http-listener-1(3);_TimeMillis=1664298197772;_LevelValue=800;|
  NestedServiceBean::doWork() called|#]

[#|2022-09-27T12:03:17.773-0500|INFO|Payara 6.2022.1.Alpha5-SNAPSHOT||_ThreadID=131;_ThreadName=http-thread-pool::http-listener-1(3);_TimeMillis=1664298197773;_LevelValue=800;|
  NestedServiceBean:: testTransactionalScope value is 1664298130350|#]

[#|2022-09-27T12:04:24.385-0500|WARNING|Payara 6.2022.1.Alpha5-SNAPSHOT|com.hazelcast.internal.cluster.impl.ClusterHeartbeatManager|_ThreadID=228;_ThreadName=hz.suspicious_bouman.cached.thread-29;_TimeMillis=1664298264385;_LevelValue=900;|
  [172.21.41.173]:4900 [development] [5.1.1] Resetting heartbeat timestamps because of huge system clock jump! Clock-Jump: 43667 ms, Heartbeat-Timeout: 60000 ms|#]

[#|2022-09-27T12:04:26.968-0500|INFO|Payara 6.2022.1.Alpha5-SNAPSHOT||_ThreadID=131;_ThreadName=http-thread-pool::http-listener-1(3);_TimeMillis=1664298266968;_LevelValue=800;|
  **TestTransactionalScopeBean destroyed**. Value is 1664298130350|#]`

### Testing Environment

Ubuntu 20.04, Azul JDK 11, Maven 3.8.6

### CODE REVIEW

**Comment by breakponchito** [comment]
Jenkins test please

File: appserver/web/weld-integration/src/main/java/org/glassfish/weld/services/InjectionServicesImpl.java [hunk 1]
```diff
@@ -138,12 +149,26 @@ public <T> void aroundInject(InjectionContext<T> injectionContext) {
...
+                            .equals("org.glassfish.cdi.transaction.TransactionalExtension")
+                            && componentEnv == null) {
+                        injectionContext.proceed();
+                        return;
+                    }
+                }
+
+                if (componentEnv == null) {
+                    //throw new IllegalStateException("No valid EE environment for injection of " + targetClassName);
+                    System.err.println("No valid EE environment for injection of " + targetClassName);
```
Comment (svendiedrichsen) [inline appserver/web/weld-integration/src/main/java/org/glassfish/weld/services/InjectionServicesImpl.java:168]: How about using a logger here?

Comment (breakponchito) [reply appserver/web/weld-integration/src/main/java/org/glassfish/weld/services/InjectionServicesImpl.java]: already added, thanks for the comment

**Comment by breakponchito** [comment]
Jenkins test please

**Comment by breakponchito** [comment]
Jenkins test please

**Comment by breakponchito** [comment]
Jenkins test please

**Comment by breakponchito** [comment]
Jenkins test please

**Comment by breakponchito** [comment]
Jenkins test please

**Comment by breakponchito** [comment]
Jenkins test please

**Comment by breakponchito** [comment]
Jenkins test please

File: appserver/web/weld-integration/src/main/java/org/glassfish/weld/services/InjectionServicesImpl.java [hunk 2]
```diff
@@ -138,12 +152,28 @@ public <T> void aroundInject(InjectionContext<T> injectionContext) {
...
+                    BackedAnnotatedType backedAnnotatedType = ((BackedAnnotatedType) annotatedType);
+                    //added condition to skip printing logs when the TransactionScopedCDIEventHelperImpl is tried
+                    // to be used for the TransactionalScoped CDI Bean
+                    if (backedAnnotatedType != null
+                            && backedAnnotatedType.getIdentifier() != null
+                            && (backedAnnotatedType.getIdentifier().getBdaId()
+                            .equals("org.glassfish.cdi.transaction.TransactionalExtension")
+                            || backedAnnotatedType.getIdentifier().getBdaId()
+                            .equals("org.glassfish.cdi.transaction.TransactionScopedContextExtension"))
+                            && componentEnv == null) {
```
Comment (Pandrex247) [inline appserver/web/weld-integration/src/main/java/org/glassfish/weld/services/InjectionServicesImpl.java:165]: Complicated if!

**Review by Pandrex247 — COMMENTED** [review]
The reproducer attached to the issue doesn't appear to work?

**Comment by breakponchito** [comment]
Jenkins test please

File: appserver/web/weld-integration/src/main/java/org/glassfish/weld/services/InjectionServicesImpl.java [hunk 3]
```diff
@@ -138,12 +152,28 @@ public <T> void aroundInject(InjectionContext<T> injectionContext) {
...
             } else {
-              if( componentEnv == null ) {
-                //throw new IllegalStateException("No valid EE environment for injection of " + targetClassName);
-                System.err.println("No valid EE environment for injection of " + targetClassName);
-                injectionContext.proceed();
-                return;
-              }
+                if (annotatedType instanceof BackedAnnotatedType) {
+                    BackedAnnotatedType backedAnnotatedType = ((BackedAnnotatedType) annotatedType);
+                    //added condition to skip printing logs when the TransactionScopedCDIEventHelperImpl is tried
```
[the remark this answers is not in the document]
Comment (breakponchito) [reply appserver/web/weld-integration/src/main/java/org/glassfish/weld/services/InjectionServicesImpl.java]: already fixed, thanks

**Comment by breakponchito** [comment]
Jenkins test please

**Comment by breakponchito** [comment]
Jenkins test please

**Review by Pandrex247 — APPROVED** [review]
Automating the reproducer as a test would be nice, though I appreciate scraping logs for the ommision of a FINE level log isn't exactly a "precise" test, so approving as-is.

**Pull request merged**